lib/DelegatedAdminsMiddleware.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/**
+ * @author Cyrille Bollu <cyr.debian@bollu.be>
+ *
+ * GroupFolders
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\GroupFolders;
+
+use OCA\GroupFolders\Service\DelegationService;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\AppFramework\Http\Response;
+use OCP\AppFramework\Http\TemplateResponse;
+use OCP\AppFramework\Middleware;
+use OCP\IRequest;
+use Psr\Log\LoggerInterface;
+
+class DelegatedAdminsMiddleware extends Middleware {
+
+    /** @var DelegationService */
+    private $delegationService;
+
+    /** @var IRequest */
+    private $request;
+
+    /**
+     *
+     * @param string $appName
+     * @param IConfig $config
+     * @param IgroupManager $groupManager
+     * @param IUserSession $userSession
+     *
+     */
+    public function __construct(IRequest $request,
+    		DelegationService $delegationService) {
+		$this->delegationService = $delegationService;
+		$this->request = $request;
+    }
+
+    /**
+     *
+     * {@inheritDoc}
+     * @see \OCP\AppFramework\Middleware::beforeController()
+     *
+     * Throws an error when the user is not allowed to use the app's APIs
+     *
+     */
+	public function beforeController($controller, $methodName) {
+
+        // method 'aclMappingSearch' implements its own access control
+        if ($methodName !== 'aclMappingSearch') {
```

Comment:
I think it would be nice to move the logic if the middleware should check the permission to a separate annotation like `@RequireGroupFolderAdmin` which can then be used on the controller methods to directly indicate there that the permissions would be checked.

See https://github.com/nextcloud/server/blob/2a054e6c04e0a40421510eb889cbf59f153c5177/lib/private/AppFramework/Middleware/Security/BruteForceMiddleware.php#L77-L80 for some inspriration on how to check that.

Reply:
ok. done